src/tracker-component.js

```diff
@@ -107,28 +108,33 @@ const setRandomUserIdCookie = () : void => {
 
 const composeCart = (type, data) => {
     // Copy the data so we don't modify it outside the scope of this method.
-    const _data = { ...data };
+    let _data = { ...data };
 
     // Devnote: Checking for cookie for backwards compatibility (the cookie check can be removed
     // a couple weeks after deploy because any cart cookie storage will be moved to localStorage
     // in this function).
-    const storedCart = window.localStorage.getItem(storage.paypalCrCart) || getCookie(storage.paypalCrCart);
+    const storedCart = window.localStorage.getItem(storage.paypalCrCart) || getCookie(storage.paypalCrCart) || '{}';
     const expiry = window.localStorage.getItem(storage.paypalCrCartExpiry);
+    const cart = JSON.parse(storedCart);
+    const currentItems = cart ? cart.items : [];
 
     if (!expiry) {
         window.localStorage.setItem(storage.paypalCrCartExpiry, Date.now() + sevenDays);
     }
 
-    if (type === 'add' && storedCart) {
-        const cart = JSON.parse(storedCart);
-        const currentItems = cart && cart.items;
-
-        if (currentItems && currentItems.length) {
-            _data.items = [
-                ...currentItems,
-                ...data.items
-            ];
-        }
+    switch (type) {
+    case 'add':
+        _data.items = addToCart(data.items, currentItems);
+        break;
+    case 'set':
+        _data.items = data.items;
+        break;
+    case 'remove':
+        _data = { ...cart, ...data };
+        _data.items = removeFromCart(data.items, currentItems);
+        break;
+    default:
+        throw new Error('invalid cart action');
```

Comment:
_*nitpick*_ its only our code calling reaching this code. do we still want to throw? we can have default just `return` instead

Reply:
throw might break merchant site.

Reply:
Calling this method with a string other than those three would be a programmer error that would break things anyway. I don't see the issue with being explicit about what went wrong.